src/MainView.vala

```diff
@@ -62,30 +65,47 @@ public class DateTime.MainView : Gtk.Grid {
 
         var time_zone_label = new Gtk.Label (_("Time Zone:"));
         time_zone_label.xalign = 1;
+        time_zone_label.valign = Gtk.Align.START;
 
         auto_time_zone_icon = new Gtk.Image ();
         auto_time_zone_icon.gicon = new ThemedIcon ("location-inactive-symbolic");
+        auto_time_zone_icon.margin_start = 6;
         auto_time_zone_icon.pixel_size = 16;
 
-        var auto_time_zone_button = new Gtk.ToggleButton ();
-        auto_time_zone_button.image = auto_time_zone_icon;
-        auto_time_zone_button.tooltip_text = _("Automatically update time zone");
+        var auto_time_zone_icon_context = auto_time_zone_icon.get_style_context ();
+        auto_time_zone_icon_context.add_class ("auto-timezone-label");
+        auto_time_zone_icon_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
-        var css_provider = new Gtk.CssProvider ();
-        css_provider.load_from_resource ("/io/elementary/switchboard/plug/datetime/Plug.css");
+        var auto_time_zone_switch_label = new Gtk.Label (_("Based on your Location:"));
+        auto_time_zone_switch_label.margin_end = 6;
+
+        var auto_time_zone_switch = new Gtk.Switch ();
+        auto_time_zone_switch.margin_top = auto_time_zone_switch.margin_bottom = 6;
+        auto_time_zone_switch.tooltip_text = _("Automatically updates the time zone when activated");
+
+        var auto_time_zone_switch_context = auto_time_zone_switch.get_style_context ();
+        auto_time_zone_switch_context.add_class ("auto-timezone");
+        auto_time_zone_switch_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+
+        var auto_time_zone_grid = new Gtk.Grid ();
+        auto_time_zone_grid.column_spacing = 6;
+        auto_time_zone_grid.hexpand = true;
+        auto_time_zone_grid.attach (auto_time_zone_icon, 0, 0);
+        auto_time_zone_grid.attach (auto_time_zone_switch_label, 1, 0);
+        auto_time_zone_grid.attach (auto_time_zone_switch, 2, 0);
 
-        var auto_time_zone_button_context = auto_time_zone_button.get_style_context ();
-        auto_time_zone_button_context.add_class ("auto-timezone");
-        auto_time_zone_button_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        var auto_time_zone_grid_context = auto_time_zone_grid.get_style_context ();
+        auto_time_zone_grid_context.add_class ("auto-timezone-grid");
+        auto_time_zone_grid_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
-        time_zone_button = new TimeZoneButton ();
-        time_zone_button.hexpand = true;
-        time_zone_button.request_timezone_change.connect (change_tz);
+        time_zone_picker = new DateTime.TimeZoneGrid ();
+        time_zone_picker.hexpand = true;
+        time_zone_picker.request_timezone_change.connect (change_tz);
 
         var time_zone_grid = new Gtk.Grid ();
```

Comment:
This grid doesn't seem to serve a purpose; just add the frame directly to the `time_zone_picker`.

```suggestion
        time_zone_picker.get_style_context ().add_class ("frame");
```